src/Classes/ServiceAPI/MyRadio_Timeslot.php

```diff
@@ -248,6 +248,24 @@ public function getTimeslotAfter($filter = [1])
         }
     }
 
+    /**
+    * Returns the currently selected timeslot (from the navbar).
+    *
+    * @return array  Time, id If null, no timeslot is selected/user is logged out.
+    */
+    public static function getUserSelectedTimeslot() {
+        if (isset($_SESSION['timeslotid'])) {
+            $timeslot = self::getInstance($_SESSION['timeslotid']);
+
+            $result = [
+                "time" => $timeslot->getStartTime(),
+                "timeslotid" => $timeslot->getId()
```

Comment:
Worth maybe returning the complete data about the timeslot (title, credits etc.), instead of just the ID? Could save an extra API call in the future.

Reply:
I don't really like the idea of returning the entire object, since this will be user-facing and could maybe return more info than expected. Hence the more limited return...

Reply:
I've added show title now though.

Reply:
It's already user-facing though (e.g. https://ury.org.uk/api/v2/timeslot/146667)

Reply:
hmm